doc/source/data/examples/batch_inference_object_detection.ipynb

```diff
@@ -0,0 +1,350 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Batch Image Object Detection using Ray Data\n",
+    "\n",
+    "In this example, we will demostrate how to use Ray Data to do batch object detection with a PyTorch model on a large set of images.\n",
+    "\n",
+    "## Vanilla Pyorch exmaple\n",
```

Comment:
Can we follow a similar structure as https://docs.google.com/document/d/19-URNhdQU97S9fdKbB01T4xNGsKyoKkM8yIZqRlm-SQ/edit#heading=h.awppmn78wn5r?

So concretely:
1. Use `read_images` to read the Dataset from S3
2. Use `take_batch` to get a batch of data. Run the full inference on that batch, and then visualize the bounding boxes of the result.
3. Scale up the logic from step 2 to the entire dataset.

cc @zhe-thoughts

Reply:
I've restructured this notebook. But there is a small difference. I'd like to show a pure PyTorch inference example first, before "read_images".  Because a previous feedback is that the reader doesn't get the purpose of the example. I think this structure may be more helpful.